Docs for Application Default Credentials, links to non-dwdoa setup (#21453)
hashicorp/vault: website/content/docs/auth/jwt/oidc-providers/google.mdx

To set up the Google-specific handling, you'll need:

- A Google Workspace account with the [super admin role](https://support.google.com/a/answer/2405986?hl=en)
  for granting domain-wide delegation API client access, or a service account that has been granted
  [the necessary](https://cloud.google.com/identity/docs/how-to/setup#auth-no-dwd) group admin roles.
- The ability to create a service account in [Google Cloud Platform](https://console.developers.google.com/iam-admin/serviceaccounts).
- To enable the [Admin SDK API](https://console.developers.google.com/apis/api/admin.googleapis.com/overview).
- An OAuth 2.0 application with an [internal user type](https://support.google.com/cloud/answer/10311615#user-type).
  We **do not** recommend using an external user type since it would allow _any user_ with a
  Google account to authenticate with Vault.

> The Google-specific handling that's used to fetch Google Workspace groups and user information in Vault uses either
> Google Workspace Domain-Wide Delegation of Authority for authentication and authorization, or group admin roles granted to a GCP service account.

Reviewer: ```suggestion
Vault uses one of the following methods to fetch Google Workspace groups and user information:

- Google Workspace Domain-Wide Delegation of Authority for authentication and authorization
- Group admin roles granted to a GCP service account.
```

Suggest simplifying to make the text easier to scan and parse.

Links to steps for setting up authentication and authorization:

> - [DWDoA](https://developers.google.com/workspace/guides/create-credentials#service-account)

Reviewer 2: ```suggestion
- [With Domain-Wide Delegation of Authority](https://developers.google.com/workspace/guides/create-credentials#service-account)
```

> - [Without DWDoA](https://cloud.google.com/identity/docs/how-to/setup#auth-no-dwd)

Reviewer 2: ```suggestion
- [Without Domain-Wide Delegation of Authority](https://cloud.google.com/identity/docs/how-to/setup#auth-no-dwd)
```

> In **step 11** within the section titled

Reviewer 2: ```suggestion
In **step 5e** within the section titled
```

- `fetch_groups` `(bool: false)` - If set to true, groups will be fetched from Google Workspace.
- `fetch_user_info` `(bool: false)` - If set to true, user info will be fetched from Google Workspace using the configured [user_custom_schemas](#user_custom_schemas).
- `groups_recurse_max_depth` `(int: <optional>)` - Group membership recursion max depth. Defaults to 0, which means don't recurse.
- `user_custom_schemas` `(string: <optional>)` - Comma-separated list of Google Workspace [custom schemas](https://developers.google.com/admin-sdk/directory/v1/guides/manage-schemas).
  Values set for Google Workspace users using custom schema fields will be fetched and made available as claims that can be used with [claim_mappings](/vault/api-docs/auth/jwt#claim_mappings). Required if [fetch_user_info](#fetch_user_info) is set to true.
- `impersonate_principal` `(string: <optional>)` - Service account email that has been granted domain-wide delegation of authority in Google Workspace.
  Required if accessing the Google Workspace Directory API through domain-wide delegation of authority, without using a service account key.

>   The service account vault is running under must be granted the `iam.serviceAccounts.signJwt` permission on this service account.

Reviewer 2: ```suggestion
  The service account Vault is running under must be granted the `iam.serviceAccounts.signJwt` permission on this service account.
```
